apps/wallet/src/ui/app/shared/faucet/useFaucetRateLimiter.ts

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useEffect, useState } from 'react';
+
+const FAUCET_RATE_LIMIT_EXPIRY_TIME_KEY = 'faucet_rate_limit_expiry_time';
+
+// We'll rate limit users for 20 minutes which should
+// more-or-less mock how the faucet API rate limits users
+const rateLimitExpiryTime = 20 * 60 * 1000;
+
+export function useFaucetRateLimiter() {
+    const [isRateLimited, setRateLimited] = useState(() => {
+        const expiryTime = localStorage.getItem(
+            FAUCET_RATE_LIMIT_EXPIRY_TIME_KEY
+        );
+        const currTime = new Date().getTime();
+        return currTime <= Number(expiryTime);
+    });
+
+    const rateLimit = () => {
+        const currTime = new Date().getTime();
```

Comment:
`Date.now()` is shorter but no need to change before merging

Reply:
I knew there was a shorter way to write this but couldn't remember off the top of my head haha